max_solutions: 3

# Planning group and link names
arm_group_name: "franko_fr3_arm"
#eef_name: "hand"
eef_name: "franko_hand"
#hand_group_name: "hand"
hand_group_name: "franko_hand"
# hand_frame: "panda_link8"
#hand_frame: "franko_fr3_hand"
hand_frame: "franko_fr3_hand_tcp"

# Poses
hand_open_pose: "open"
hand_close_pose: "close"
arm_home_pose: "franko_home"

# Scene frames
world_frame: "franko_world"
table_reference_frame: "franko_world"
# object_reference_frame: "world"
object_reference_frame: "franko_fr3_link0"
magazine_reference_frame: "franko_environment"
surface_link: "table"

# Collision object for picking
# CYLINDER object specifications
objectT1_name: "objectT1"
objectT_dimensions: [0.075, 0.0125]
objectT1_file: "/home/humation/Documents/177688_CQT-15.stl"
objectT1_pose: [0.2825, 0.367, 0.06, 1.5708, 0, 0]

objectT2_name: "objectT2"
objectT2_file: "/home/humation/Documents/177688_CQT-15.stl"
objectT2_pose: [0.1755, 0.37, 0.06, 1.5708, 0, 0]

objectT3_name: "objectT3"
objectT3_file: "/home/humation/Documents/177688_CQT-15.stl"
objectT3_pose: [0.336, 0.425, 0.06, 1.5708, 0, 0]

objectT4_name: "objectT4"
objectT4_file: "/home/humation/Documents/177688_CQT-15.stl"
objectT4_pose: [0.229, 0.428, 0.06, 1.5708, 0, 0]

# ---

objectL1_name: "objectL1"
objectL1_file: "/home/humation/Documents/177692_CQL-15.stl"
objectL1_pose: [0.286, 0.545, 0.06, 1.5708, 0, 0]

objectL2_name: "objectL2"
objectL2_file: "/home/humation/Documents/177692_CQL-15.stl"
objectL2_pose: [0.176, 0.549, 0.06, 1.5708, 0, 0]

objectL3_name: "objectL3"
objectL3_file: "/home/humation/Documents/177692_CQL-15.stl"
objectL3_pose: [0.338, 0.605, 0.06, 1.5708, 0, 0]

objectL4_name: "objectL4"
objectL4_file: "/home/humation/Documents/177692_CQL-15.stl"
objectL4_pose: [0.232, 0.606, 0.06, 1.5708, 0, 0]

objectI_name: "objectI"
#objectI_dimensions: [0.043, 0.0125] # [height, radius]
objectI_file: "/home/humation/Documents/177696_CQ-15.stl"
objectI_pose: [0,2175, 0.546, 0.057, 0, 0, 0]
#objectI_pose: [0.1425, 0.6525, 0.057, 0, 0, 0]

# Collision object for assembly
# CYLINDER object specifications
assembly_object_name: "assembly_object"
assembly_object_dimensions: [0.2, 0.01] # [height, radius]
#assembly_object_pose: [0.1, -0.5, 0.376, 1.5708, 0, 0]
assembly_object_pose: [0.0, 0.0, 0.02, 0, 0, 0]
assembly_object1_pose: [0.12, 0.0, -0.1, 0, 1.5708, 0]
assembly_object2_pose: [-0.12, 0.0, -0.1, 0, 1.5708, 0]

# Table model
spawn_table: true
table_name: "table"
table_dimensions: [0.75, 0.75, 0.1]  # [length, width, height]
table_pose: [0.025, 0.475, 0.0, 0, 0, 0]

spawn_magazines: true
magazine_name: "magazine"
#magazine_dimensions: [0.1, 0.2, 0.01]  # [length, width, height]
#magazine_file: "/home/humation/Documents/magazin_festo_komplett.stl"
magazine_file: "/home/humation/Documents/magazine_mirrored.stl"
magazine_poseT: [1.787, 0.145, 0.1, 0, 0, 1.571]
magazine_poseL: [1.967, 0.145, 0.1, 0, 0, 1.571]
magazineT_assembled_file: "/home/humation/Documents/magazinT_assembled.stl"
magazineT_assembled_pose: [0.93, 0.6175, 0.295, 0, 0, -1.571]
#magazine_pose: [0.11, 0.625, 0.0, 0, 0, -1.5708]

# Gripper grasp frame transform [x,y,z,r,p,y] y,p,r?!
# grasp_frame_transform: [0, 0, 0.1, 1.571, 0.785, 1.571]
# grasp_frame_transform: [0, 0, 0.1, 0, 1.571, 0]
#grasp_frame_transform: [0, 0, 0.12, 0, 3.14, 0]
#grasp_frame_transform: [0, 0, 0.12, 0, 1.571, 0]
#grasp_frame_transform: [0, 0, 0.12, -1.571, -1.571, 0]
grasp_frame_transform: [0, 0, 0, -1.571, -1.571, 0]
#assemble_frame_transform: [0, 0, 0, -1.571, 1.571, 0]
assemble_frame_transform: [0, 0, 0, 0, 0, 0]

# Place pose [x,y,z,r,p,y]
# place_pose: [0.6, -0.15, 0, 0, 0, 0]
# place_pose: [0.1, -0.65, 0.376, 0, 0, 0] #current!?
# place_pose1: [0.2, -0.75, 0.376, 0, 0, 0]
place_pose1: [0.2, -0.75, 0.376, 1.571, 0, -1.571] # bottom upright
# place_pose1: [0.2, -0.75, 0.376, 1.571, 1.571, -1.571] # bottom side
#place_pose2: [0.2, -0.75, 0.376, -1.571, -1.571, 0] 
place_pose2: [0.2, -0.75, 0.376, 0, -3.14, -1.571] # front upright
#place_pose2: [0.2, -0.75, 0.376, 0, -1.571, 0, 0] 
#place_pose3: [0.2, -0.75, 0.376, 1.571, 1.571, 0] 
#place_pose3: [0.2, -0.75, 0.376, -3.141, 0.000, 1.658] 
#place_pose3: [0.2, -0.75, 0.376, -1.571, 4.71, 1.571] 
#place_pose3: [0.2, -0.75, 0.376, -1.571, 1.571, 3.14]
place_pose3: [0.2, -0.75, 0.376, 0, -3.14, 1.571] # back upright

#(-0.14 + (object_id * 0.07), -0.465, 0.377, 0.78)
assembly_place_pose: [0.93, 0.53, 0.3725, 0, 3.14, 0.0] # back upright

assembled_pipe1_pose: [-0.147909, -0.583942, 0.172329,  1.571, 0, 0]
assembled_pipe2_pose: [-0.147926, -0.343943, 0.172775,  1.571, 0, 0]
assembled_pipe3_pose: [-0.147929, -0.46372, 0.052552,  0, 0, 0]
assembled_T_pose: [-0.148368, -0.46444, 0.272957,  1.571, 0, 0]

# place_pose: [0.5, 0.3, 0.532, 0, 0, 0]
place_surface_offset: 0.0001 # place offset from table

# Valid distance range when approaching an object for picking
approach_object_min_dist: 0.1
approach_object_max_dist: 0.15

# Valid height range when lifting an object after pick
lift_object_min_dist: 0.05
lift_object_max_dist: 0.25

# Valid height range when lifting an object after pick
move_home_object_min_dist: 0.01
move_home_object_max_dist: 0.16
